### ISSUE TITLE
Bug: Segmentation violation when trying to run latest (7a69ddb) on both Pi 2b and amd64 

### ISSUE THREAD

**Issue by distbit0** [mxpv/podsync#322]
Hi I just downloaded the latest podsync source and compiled it using env GOOS=linux GOARCH=arm GOARM=7 for my pi and without any flags (in a separate session) for amd64 ubuntu laptop. I however experienced the same issue when running it with my config.toml file on both machines:

```
pimania@distPI:~/Dev/ytToRSS$ panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x7143dc]

goroutine 70 [running]:
github.com/mxpv/podsync/services/update.(*Manager).Update(0x2894330, {0xaddbe4, 0x2894360}, 0x0)
	/home/pimania/Downloads/podsync-main/services/update/updater.go:59 +0x40
main.main.func3()
	/home/pimania/Downloads/podsync-main/cmd/podsync/main.go:179 +0xfc
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/pimania/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:57 +0x68
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/pimania/go/pkg/mod/golang.org/x/sync@v0.0.0-20210220032951-036812b2e83c/errgroup/errgroup.go:54 +0x88
```

I have been running podsync on my pi largely without any issues for about 3 months so far.

My config file is attached (without keys/url):
[config (copy).zip](https://github.com/mxpv/podsync/files/8904461/config.copy.zip)

**Comment by Th0masL** [comment]
Yeah, I was trying to build from source too in order to be able to benefit from the PR we merged about fixing the 50 page_size limit, and I have exactly the same problem.

So it looks like there's something broken in the `main` branch.

**Comment by distbit0** [comment]
Glad it is not just me!

**Comment by Th0masL** [comment]
I'm suspecting the `feat/s3` PR to be the reason of the problem, because if I change my config file to try to start using the new syntax I get another problem where `podsync` can't even find the right value for my `data` folder :

```
time="2022-06-15T00:49:20Z" level=fatal msg="failed to load configuration file" error="1 error occurred:\n\t* data directory is required for local storage\n\n"
```

```
[server]
port = 8080
hostname = "https://<my_server_hostname>"
#data_dir = "/app/data" # <=== Recently commented out

# <=== Recently Added
[storage]
  type = "local"
  [storage.local]
  data_dir = "/app/data"
```

**Comment by Contextualist** [comment]
Thanks for reporting, @distbit0 and @Th0masL! Each of you actually identify a different regression from #317. I just pushed the fixes in #324. Could you test to see if it work?

**Comment by distbit0** [comment]
Yeah it is now working :+1: Epic
